lib/EnvironmentPlugin.js

```diff
@@ -47,7 +57,11 @@ class EnvironmentPlugin {
 					);
 
 					error.name = "EnvVariableNotDefinedError";
-					compilation.warnings.push(error);
+					if (this.options["errorLevel"] === "error") {
```

Comment:
I think this should be a breaking change instead and always use errors. Warnings doesn't really make sense. Either build need the environment variable (which should error if not provided) or it doesn't need it (in which case you need to provide a default value).

In this case we don't need the options argument.

Reply:
Makes sense to me, I'll make the changes